File: BE/src/main/java/com/ssafy/e106/domain/subscription/dto/response/CheckinHistoryItemResponse.java
package com.ssafy.e106.domain.subscription.dto.response;

import java.time.LocalDateTime;

import com.ssafy.e106.domain.subscription.entity.CheckinRecord;

public record CheckinHistoryItemResponse(
    Long checkinRecordId,
    Long serviceId,
    String serviceName,
    String cycleYm,
    String response,
    LocalDateTime respondedAt) {

  public static CheckinHistoryItemResponse from(CheckinRecord checkinRecord) {
    return new CheckinHistoryItemResponse(
        checkinRecord.getCheckinRecordId(),
        checkinRecord.getService().getServiceId(),
        checkinRecord.getService().getName(),
        checkinRecord.getCycleYm(),
        checkinRecord.getResponse().name(),
        checkinRecord.getRespondedAt());
  }
}

File: FE/app/src/main/java/com/ssafy/e106/data/dto/subscription/CheckInHistoryItemResponse.kt
package com.ssafy.e106.data.dto.subscription

import kotlinx.serialization.Serializable

@Serializable
data class CheckInHistoryItemResponse(
    val checkinRecordId: Long,
    val serviceId: Long,
    val serviceName: String,
    val cycleYm: String,
    val response: String,
    val respondedAt: String,
)
